src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -151,6 +160,16 @@ class BlockchainImpl(
 
   override def getTotalDifficultyByHash(blockhash: ByteString): Option[BigInt] = totalDifficultyStorage.get(blockhash)
 
+  override def getAccount(address: Address, blockNumber: BigInt): Option[Account] =
+    getBlockHeaderByNumber(blockNumber).flatMap { bh =>
+      val mpt = new MerklePatriciaTrie[Address, Account](
```

Comment:
Should we use something similar to https://github.com/input-output-hk/etc-client/blob/master/src/main/scala/io/iohk/ethereum/ledger/InMemoryWorldStateProxy.scala#L96 and use `HashByteArraySerializable` so that the addresses are hashed before using them to access the MPT? It should be just adding the use of `(HashByteArraySerializable(addressSerializer), accountSerializer)` or maybe extracting this in some way (as it is shared with `InMemoryWorldStateProxy`).